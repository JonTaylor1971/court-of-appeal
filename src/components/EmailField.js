import React, { Component } from 'react';

class EmailField extends Component {

    constructor(props) {
        super(props);

        this.handleFieldChange = this.handleFieldChange.bind(this);
        this.validate = this.validate.bind(this);
        this.state = {
            size: props.value  && props.value.length > 30 ? props.value.length : 30,
            className: 'form-field'
        };
        this.regex = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
    }

    componentDidMount() {
        if (this.props.value) {
            this.validate();
        }
    }

    render() {
        let autofocus = this.props.autofocus ? true : false;

        return (<input
            id={this.props.id}
            name={this.props.name}
            autoFocus={autofocus}
            ref={this.props.ref}
            value={this.props.value || ''}
            onChange={this.handleFieldChange}
            placeholder={this.props.placeholder}
            size={this.state.size}
            className={this.state.className}
            maxLength={254}
            readOnly={this.props.readOnly}
            onBlur={this.validate}
        />);
    }

    handleFieldChange(e) {
        let size = this.props.value.length > 30 ? this.props.value.length  : 30;
        this.setState({size: size});
        if (!this.props.readOnly) {
            this.props.handleFieldChange(e);
        }
    }

    validate() {
        if (!this.props.value || this.isValidEmailAddress())
        {
            this.setState({className: "form-field", title: ""}, this.props.validate(true));
        } else {
            this.setState({className: "form-field invalid-field", title: "ERROR: Invalid Phone Number"}, this.props.validate(false));
        }
    }

    isValidEmailAddress() {
        return this.props.value.match(this.regex) != null;
    }


}
export default EmailField;

