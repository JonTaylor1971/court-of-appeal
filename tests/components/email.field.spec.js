import React from 'react';
import EmailField from '../../src/components/EmailField';
import renderer from 'react-test-renderer';

test('should only allow numeric, dash, and parentheses entries', ()=> {

    let value = '';
    let handleFieldChange = (e) => { value = e.target.value; };

    const phoneField = renderer.create(
        <EmailField id="number-field" value={value}
                    handleFieldChange={handleFieldChange.bind(this)}/>,
    );
    let tree = phoneField.toJSON();
    expect(tree).toMatchSnapshot();

    let e = { target: { value: "42"}};
    tree.props.onChange(e);
    expect(value).toMatch('42');

    e = { target: { value: "bah" } };
    tree.props.onChange(e);
    expect(value).toMatch('');

    e = { target: { value: "(Johnny)JOHNsmiath@john.com" } };
    tree.props.onChange(e);
    expect(value).toMatch('(Johnny)JOHNsmiath@john.com');

    e = { target: { value: "\"\\\"slashy@pinki.ca" } };
    tree.props.onChange(e);
    expect(value).toMatch('slashy@pinki.ca');

    e = { target: { value: "   " } };
    tree.props.onChange(e);
    expect(value).toMatch('');
});
