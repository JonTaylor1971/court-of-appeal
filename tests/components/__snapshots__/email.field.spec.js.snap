// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only allow numeric, dash, and parentheses entries 1`] = `
<input
  autoFocus={false}
  className="form-field"
  id="number-field"
  maxLength={254}
  name={undefined}
  onBlur={[Function]}
  onChange={[Function]}
  placeholder={undefined}
  readOnly={undefined}
  size={30}
  value=""
/>
`;
